server/document.js

```diff
@@ -113,7 +113,12 @@ export class Head extends Component {
       {this.getPreloadMainLinks()}
       {this.getCssLinks()}
       {styles || null}
-      {this.props.children}
+      {process.env.NODE_ENV === 'development' && (React.Children.toArray(this.props.children)).map((child) => {
```

Comment:
Am I missing something or `this.props.children` won't be rendered at all if `NODE_ENV` is not `production` ? 😅

Reply:
Correct, this is wrong, and uglify won't remove the block if it's not development. Instead we have to do something like this:

```
let children
if (process.env.NODE_ENV === 'development') {
 children = // etc etc
} else {
 children = this.props.children
}
```

This way uglify will remove the specific block

Reply:
Why not just use `React.Children.forEach` to show the warning ? And keep `{this.props.children}` ?
```js
if (process.env.NODE_ENV === 'development') {
  React.Children.forEach(this.props.children, (child) => {
    if (child.type === 'title') {
      console.warn("...")
    }
  })
}
```

Reply:
Doing 1 loop instead of 2 🤔

Reply:
But only in `development` 🙈

Reply:
Random question - is it not better practice also to check process.env.NODE_ENV !== 'production' instead? I thought that it was safer to assume that 'production' would be present in prod. Some people might want warnings to show in 'staging' also for example.

Reply:
Yes ! That's also how it was implemented 🙂

https://github.com/zeit/next.js/pull/5160#discussion_r219099014